src/renderer/entities/network/model/network-model.ts

```diff
@@ -174,77 +264,266 @@ sample({
+  fn: (subscriptions, unsubscribes) => {
+    unsubscribes.forEach(([chainId, cb]) => {
+      subscriptions[chainId as ChainId] = cb;
+    });
+
+    return subscriptions;
+  },
+  target: $metadataSubscriptions,
+});
+
+// =====================================================
+// =============== Connection section ==================
+// =====================================================
+
+$connections
+  .on(populateConnectionsFx.doneData, (_, connections) => {
+    return keyBy(connections, 'chainId');
+  })
+  .on(createConnectionFx.doneData, (state, connection) => {
+    return connection
+      ? {
+          ...state,
+          [connection.chainId]: connection,
+        }
+      : state;
+  })
+  .on(deleteConnectionFx.doneData, (state, connectionId) => {
+    const deletedConnection = Object.values(state).find((c) => c.id === connectionId);
+
+    if (!deletedConnection?.chainId) return state;
+
+    const { [deletedConnection.chainId]: _, ...newConnections } = state;
+
+    return newConnections;
+  })
+  .on(updateConnectionFx.doneData, (state, connection) => {
+    return {
+      ...state,
+      [connection.chainId]: connection,
+    };
+  });
+
+forward({
+  from: kernelModel.events.appStarted,
+  to: populateConnectionsFx,
+});
+
+sample({
+  clock: rpcNodeAdded,
+  source: $connections,
+  fn: (connections, { chainId, rpcNode }) => {
+    const connection = connections[chainId];
+
+    connection.customNodes = (connection.customNodes || []).concat(rpcNode);
+    connection.activeNode = rpcNode;
+
+    return connection;
+  },
+  target: updateConnectionFx,
+});
+
+sample({
+  clock: rpcNodeUpdated,
+  source: $connections,
+  filter: (_, { oldNode }) => !!oldNode,
+  fn: (connections, { chainId, oldNode, rpcNode }) => {
+    const connection = connections[chainId];
+
+    connection.customNodes = (connection.customNodes || []).map((node) => {
+      if (node.url === oldNode!.url && node.name === oldNode!.name) return rpcNode;
+
+      return node;
+    });
+
+    return connection;
+  },
+  target: updateConnectionFx,
+});
+
+sample({
+  clock: rpcNodeRemoved,
+  source: $connections,
+  fn: (connections, { chainId, rpcNode }) => {
+    const connection = connections[chainId];
+
+    connection.customNodes = (connection.customNodes || []).filter((node) => node.url !== rpcNode.url);
+
+    return connection;
+  },
+  target: updateConnectionFx,
+});
+
+sample({
+  clock: updateConnectionFx.done,
+  source: $chains,
+  filter: (chains, { params: connection }) => {
+    const activeNode = connection.activeNode;
+    const nodes = (connection.customNodes || []).concat(chains[connection.chainId].nodes || []);
+    const isNodeFound = nodes.some((node) => node.url === activeNode?.url);
+
+    return !isNodeFound;
+  },
+  fn: (chains, { params: connection }) => ({
+    ...connection,
+    activeNode: chains[connection.chainId].nodes[0],
   }),
-  target: disconnectFx,
+  target: updateConnectionFx,
```

Comment:
Does it mean that we will switch from light client to first websocket node it some custom websocket node will be added/updated/deleted?